### PR TITLE
Implemented cluster API creation/deletion/update test cases

### PR DESCRIPTION
**Pull request by kimeunju108**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
This PR proposes unit test cases of cluster controller

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
NONE

### CODE REVIEW

**Review by jshaofuturewei — APPROVED** [review]

**Review by CoderKevinZhang — COMMENTED** [review]
I have a question. Were you sure that your unit tests had been run? Did you update the make file to let the Jenkins run your tests under globalscheduler folder

**Comment by kimeunju108** [comment]
> I have a question. Were you sure that your unit tests had been run? Did you update the make file to let the Jenkins run your tests under globalscheduler folder

Yes, it performed unit test and also passed CICD test.

**Review by CoderKevinZhang — APPROVED** [review]

File: globalscheduler/controllers/cluster/cluster_controller.go [hunk 1]
```diff
@@ -210,22 +212,63 @@ func (c *ClusterController) processNextWorkItem() bool {
...
+	fmt.Println("key, namespace, clusterName: %v,%v,%v", key, namespace, clusterName)
 	cluster, err := c.clusterlister.Clusters(namespace).Get(clusterName)
 	if err != nil || cluster == nil {
 		klog.Errorf("Failed to retrieve cluster in local cache by cluster name - %s", clusterName)
+		fmt.Println("Failed to retrieve cluster in local cache by cluster name - %s, %s", namespace, clusterName)
+		fmt.Println("Failed to retrieve cluster in local cache by cluster  - %v, %v", cluster, err)
 		return err
 	}
+	result, err := c.perform(keyWithEventType.EventType, cluster)
+	if !result {
```
Comment (cathyhongzhang) [inline globalscheduler/controllers/cluster/cluster_controller.go:234]: This code addition is not for unit test purpose. It seems a placeholder for adding cluster controller logic in the future, i.e. the gRPC communication with resource collector. Could you add some code comments here for easy understanding?

File: globalscheduler/controllers/cluster/cluster_controller.go [hunk 2]
```diff
@@ -210,22 +212,63 @@ func (c *ClusterController) processNextWorkItem() bool {
...
+	clusterCopy := cluster.DeepCopy()
+	_, err = c.clusterclientset.GlobalschedulerV1().Clusters(cluster.Namespace).Update(clusterCopy)
+	if err != nil {
+		klog.Infof("cluster update error - %v", clusterName)
+		return err
+	}
+	return err
+}
+
+func (c *ClusterController) updateClusterStatus(cluster *clusterv1.Cluster) error {
```
Comment (cathyhongzhang) [inline globalscheduler/controllers/cluster/cluster_controller.go:262]: To be consistent, could you move this function definition to the same file that defines the "createCluster" function?

File: globalscheduler/controllers/cluster/cluster_controller_test.go [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+	f := newFixture(t)
+	cluster := newCluster("cluster2")
+	f.clusterLister = append(f.clusterLister, cluster)
+	f.objects = append(f.objects, cluster)
+	f.expectDeleteClusterStatusAction(cluster)
+	f.run(getKey(cluster, t), EventType_Delete)
+}
+
+func TestClusterUnknownEvent(t *testing.T) {
+	f := newFixture(t)
```
Comment (cathyhongzhang) [inline globalscheduler/controllers/cluster/cluster_controller_test.go:334]: Suggest that you add some negative test cases, such as when the user registers a cluster with incorrect data structures or or the cluster misses some mandatory information or has conflicting info. Our code should have validation logic to handle incorrect user input/configs.

Comment (cathyhongzhang) [reply globalscheduler/controllers/cluster/cluster_controller_test.go:334]: You can add more negative UT test cases later in another PR.

**Comment by cathyhongzhang** [comment]
LGTM

**Review by cathyhongzhang — APPROVED** [review]

**Pull request merged**